server/src/test/java/org/candlepin/resteasy/DateFormatterTest.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2009 - 2021 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.resteasy;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+import org.jboss.resteasy.spi.util.FindAnnotation;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+
+import java.lang.annotation.Annotation;
+import java.util.Locale;
+
+public class DateFormatterTest {
+
+    @AfterEach
+    public void cleanup() {
+        Locale.setDefault(Locale.US);
+    }
+
+    @Test
+    public void testFromStringParseWithAustralianEnglishAsSystemLocale() {
+        Locale.setDefault(new Locale("en", "AU"));
+        DateFormatter formatter = new DateFormatter();
+        Annotation[] annotations = new Annotation[1];
+        DateFormat dateFormatAnnotation = mock(DateFormat.class);
+        try (MockedStatic<FindAnnotation> findAnnotationMock = mockStatic(FindAnnotation.class)) {
```

Comment:
We could use a test class here. That would allow us to drop the mocking and just collect annotation from the class. I think it would help readability and it would be easier to add more tests.

Something like:
```java
    @Test
    public void testFromStringParseWithAustralianEnglishAsSystemLocale() {
        Locale.setDefault(new Locale("en", "AU"));
        DateFormatter formatter = new DateFormatter();
        formatter.setAnnotations(Annotated.class.getAnnotations());

        assertDoesNotThrow(() -> formatter.fromString("Thu, 01 Jan 1970 00:00:00 GMT"));
    }

    @DateFormat("EEE, dd MMM yyyy HH:mm:ss z")
    private static class Annotated {}
```

Reply:
done